### ISSUE TITLE
Consider splitting client SDKs into separate repos?

### ISSUE THREAD

**Issue by damienpontifex** [Azure/azure-mobile-services#628]
Considering the different release cycles for each platform, I was wondering whether it would be easier having each in it's own repo?

As an example, the iOS v2.0 just released. A git tag of 2.0 wouldn't really make sense as the Android, Managed, Javascript platforms are at different versions. Speaking from the iOS side, cocoa pods could be used to distribute the SDK (like nuget for the managed SDK) but the cocoa pods spec won't allow a version of iOS-2.0.0.

**Comment by phvannor** [comment]
Yes, I agree.  We are currently splitting out -tests and -quickstarts as well.  Once those two are complete, my next goal is to get the SDK folders split (and the main repo submodules?)  

I just don't know when it will happen.  FOr now, I can put a new tag on master for CocoaPods.  Does it only allow numbers?  Any suggestions for what to use?

**Comment by damienpontifex** [comment]
I have a podspec file locally in my project as the main repo doesn't have one. Putting the tag as iOS-2.0.0 throws all kinds of errors on install about semantic versioning. I made the assumption on this that it is looking for a plain number format i.e. 2.0.0

It's no issue really (in my case) unless you support integrating CocoaPods with the releases of the iOS SDK. At this stage, I just use the head of the branch rather than the tag.

Probably can leave it until there is a standalone repo (if you go that way) and then CocoaPods integration would make things easier for the iOS developers.

**Comment by baumatron** [comment]
If you'd like, take a look at the podspec that I have checked in: https://github.com/baumatron/azure-mobile-services/blob/add_cocoapods_2.0.0_podspec/WindowsAzureMobileServices.podspec.json

I checked in the podspec to a separate branch in my fork, which actually points back to azure/azure-mobile-services. I have it pointed at master iOS-2.0.0 and it seems to work fine for me.

This may be orthogonal to the splitting up of the SDKs, but it seems that it's possible to use iOS-2.0.0 as a tag in the spec at least. Hope this is helpful.
